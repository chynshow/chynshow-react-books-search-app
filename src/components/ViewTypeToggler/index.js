import React, { useContext } from 'react';
import { AppContext } from '../../state/AppContext';
import ThListSVG from '../SVG/ThListSVG';
import ThSVG from '../SVG/ThSVG';

const ViewTypeToggler = () => {
  const { setView } = useContext(AppContext);
  return (
    <div className='c-view-type'>
      <button
        className='c-btn c-btn--primary c-view-type__btn'
        onClick={() => setView('one')}
      >
        <ThListSVG />
      </button>
      <button
        className='c-btn c-btn--primary c-view-type__btn'
        onClick={() => setView('four')}
      >
        <ThSVG />
      </button>
    </div>
  );
};

export default ViewTypeToggler;
