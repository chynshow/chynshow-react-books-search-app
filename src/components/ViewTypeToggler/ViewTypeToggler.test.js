import React from 'react';
import ViewTypeToggler from './';
import { shallow } from 'enzyme';

it('should renders ViewTypeToggler component', () => {
  const c = shallow(<ViewTypeToggler />);
  expect(c).toMatchSnapshot();
});
