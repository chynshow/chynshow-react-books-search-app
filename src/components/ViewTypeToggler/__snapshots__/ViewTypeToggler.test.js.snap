// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders ViewTypeToggler component 1`] = `
<div
  className="c-view-type"
>
  <button
    className="c-btn c-btn--primary c-view-type__btn"
    onClick={[Function]}
  >
    <ThListSVG />
  </button>
  <button
    className="c-btn c-btn--primary c-view-type__btn"
    onClick={[Function]}
  >
    <ThSVG />
  </button>
</div>
`;
